service/history/archivalQueueFactory.go

```diff
@@ -0,0 +1,140 @@
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package history
+
+import (
+	"math"
+
+	"go.uber.org/fx"
+
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/log/tag"
+	"go.temporal.io/server/common/metrics"
+	"go.temporal.io/server/service/history/archival"
+	"go.temporal.io/server/service/history/queues"
+	"go.temporal.io/server/service/history/shard"
+	"go.temporal.io/server/service/history/tasks"
+	"go.temporal.io/server/service/history/workflow"
+)
+
+const (
+	archivalQueuePersistenceMaxRPSRatio = 0.15
+)
+
+type (
+	archivalQueueFactoryParams struct {
+		fx.In
+
+		QueueFactoryBaseParams
+	}
+
+	archivalQueueFactory struct {
+		archivalQueueFactoryParams
+		QueueFactoryBase
+		archiver archival.Archiver
+	}
+)
+
+func NewArchivalQueueFactory(
+	params archivalQueueFactoryParams,
+) QueueFactory {
+	hostScheduler := queues.NewNamespacePriorityScheduler(
+		params.ClusterMetadata.GetCurrentClusterName(),
+		queues.NamespacePrioritySchedulerOptions{
+			WorkerCount:            params.Config.ArchivalProcessorSchedulerWorkerCount,
+			ActiveNamespaceWeights: params.Config.ArchivalProcessorSchedulerRoundRobinWeights,
+			// we don't need standby weights because we only run in the active cluster
+			EnableRateLimiter:           params.Config.TaskSchedulerEnableRateLimiter,
+			MaxDispatchThrottleDuration: HostSchedulerMaxDispatchThrottleDuration,
+		},
+		params.NamespaceRegistry,
+		params.SchedulerRateLimiter,
+		params.TimeSource,
+		params.MetricsHandler.WithTags(metrics.OperationTag(metrics.OperationArchivalQueueProcessorScope)),
+		params.Logger,
+	)
+	return &archivalQueueFactory{
+		archivalQueueFactoryParams: params,
+		QueueFactoryBase: QueueFactoryBase{
+			HostScheduler:        hostScheduler,
+			HostPriorityAssigner: queues.NewPriorityAssigner(),
```

Comment:
plz double check if archival task is marked as low priority.

Reply:
Done and added a test for it